src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -27,7 +27,13 @@
      */
     public function supports(Request $request, ArgumentMetadata $argument)
     {
-        return !$argument->isVariadic() && $request->attributes->has($argument->getName());
+        if ($argument->isVariadic()) {
+            return false;
+        }
+
+        $argumentName = $argument->getName();
+
+        return $request->attributes->has($argumentName) && null !== $request->attributes->get($argumentName);
```

Comment:
If you ask me, routing information should not be affected by arguments in the method that are not mapped in the route, but that's exactly what happens right now :cry: